components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -352,11 +352,25 @@ func (sb *sensorBase) determineHeadingFunc(ctx context.Context,
 // if loop is tuning, return an error
 // if loop has been tuned but the values haven't been added to the config, error with tuned values.
 func (sb *sensorBase) checkTuningStatus() error {
-	if sb.loop != nil && sb.loop.GetTuning(context.Background()) {
```

Comment:
turns out the loop is always nil at this point because we do not set the loop until tuning completes.